cava/CoqStyleGuide.md

```diff
@@ -0,0 +1,254 @@
+
+## Lemma/Theorem statements
+
+- Generally, use `Theorem` for the most important, top-level facts you prove
+  and `Lemma` for everything else.
+- Insert a line break after the colon in the lemma statement.
+- Insert a line break after the comma for `forall` or `exist` quantifiers.
+- Implication arrows (`->`) should share a line with the previous hypothesis,
+  not the following one.
+- There is no need to make a line break after every `->`; short preconditions
+  may share a line.
+
+## Proofs and tactics
+
+- Use the `Proof` command (lined up vertically with `Lemma` or `Theorem` it
+  corresponds to) to open a proof, and indent the first line after it 2 spaces.
+- Very small proofs (where `Proof. <tactics> Qed.` is <= 80 characters) can go
+  all in one line.
+- When ending a proof, align the ending statement (`Qed`, `Admitted`, etc.)
+  with `Proof`.
+- Avoid using autogenerated names (e.g. `H0`, `n0`). They make the proof
+  break when new hypotheses are added or variable names change.
+- Use curly braces `{}` for subgoals, instead of bullets.
+- *Never write tactics with more than one subgoal focused.* This can make the
+  proof very confusing to step through! If you have more than one subgoal, use
+  curly braces.
+- Consider adding a comment after the opening curly brace that explains what
+  case you're in (see example).
+  * This is not necessary for small subgoals but can help show the major lines
+    of reasoning in large proofs.
+- If invoking a tactic that is expected to return multiple subgoals, use `[ |
+  ... | ]` before the `.` to explicitly specify how many subgoals you expect.
+  * Examples: `split; [ | ].` `induction z; [ | | ].`
+  * This helps make code more maintainable, because it fails immediately if
+    your tactic no longer solves as many subgoals as expected (or unexpectedly
+solves more).
+- If invoking a string of tactics (composed by `;`) that will break the goal
+  into multiple subgoals and then solve all but one, still use `[ ]` to enforce
+that all but one goal is solved.
+  * Example: `split; try lia; [ ]`.
+- Tactics that consist only of `repeat`ing a procedure (e.g. `repeat match`,
+  `repeat first`) should factor out a single step of that procedure a separate
+tactic called `<tactic name>_step`, because the single-step version is much
+easier to debug. For instance:
+  ```coq
+  Ltac crush_step :=
+    match goal with
+    | _ => progress subst
+    | _ => reflexivity
+    end.
+  Ltac crush := repeat crush_step.
+  ```
+
+## Naming
+
+- Helper proofs about standard library datatypes should go in a module that is
+  named to match the standard library module (see example).
+  * This makes the helper proofs look like standard-library ones, which is
+    helpful for categorizing them if they're genuinely at the standard-library
+    level of abstraction.
+- Names of modules should start with capital letters.
+- Names of inductives and their constructors should start with capital letters.
```

Comment:
Going by the example perhaps add:
`- Names of other definitions/lemmas should be snake case`